ReactiveCocoa/Swift/Property.swift

```diff
@@ -240,12 +240,18 @@ public func <~ <P: MutablePropertyType>(property: P, signal: Signal<P.Value, NoE
 public func <~ <P: MutablePropertyType>(property: P, producer: SignalProducer<P.Value, NoError>) -> Disposable {
 	let disposable = CompositeDisposable()
 
-	producer.startWithSignal { signal, signalDisposable in
-		property <~ signal
-		disposable += signalDisposable
-
-		disposable += property.producer.startWithCompleted {
+	disposable += property.producer.startWithCompleted {
+		disposable.dispose()
+	}
+	
+	disposable += producer.start { [weak property] event in
```

Comment:
`property` should be `unowned`, to better reflect the ownership relationship, and so that we don't have to reason about what to do if it becomes `nil`.